crates/uv-interpreter/src/environment.rs

```diff
@@ -31,6 +34,21 @@ impl PythonEnvironment {
         } else if system.is_preferred() {
             Self::from_default_python(cache)
         } else {
+            // First check for a parent intepreter
+            match Self::from_parent_interpreter(cache) {
+                Ok(env) => {
+                    if !system.is_allowed()
+                        && env.interpreter().base_prefix() != env.interpreter().base_exec_prefix()
+                    {
+                        warn!("Allowing system interpreter without `--system` since uv was invoked from Python");
```

Comment:
Should we do a real `warn_user_once` here?

Reply:
(I think it may be overkill)

Reply:
Can you use `env.interpreter().is_virtualenv()`?

Reply:
Yeah I copied this from below and I don't know why it's not using that either. Will do.